services/mailer.test.js

```diff
@@ -0,0 +1,38 @@
+jest.mock('../lib/log')
+jest.mock('mailgun-js')
+const mailgun = require('mailgun-js');
+
+
+const sendFn = jest.fn().mockReturnValue( Promise.resolve('hello'))
+
+mailgun.mockImplementation(() => {
+    return{
+        messages: () => {
+            return {
+                send: sendFn
+            }
+        }
+    }
+})
+
+//requiring mailer uses the mailgun function so you want to initialize it after you define it at lines 8-16
+const email = require('./mailer');
```

Comment:
you should probably require `mailer` after you stub out the logger. Otherwise the logger used in your mailer will be the original, not the stubbed out one.